ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+import autodux from 'autodux';
+import { doFetch } from './utils/fetch-websocket';
+
+export const {
+  reducer,
+  initial: defaultState,
+  actions: {
+    activateConnection,
+    deactivateConnection,
+    serverConnected,
+    serverDisconnected,
+    updatePurses,
+    changePurse,
+    swapInputs,
+    changeAmount,
+    createOffer,
+    resetState,
+    updateInvitationDepositId,
+  },
+} = autodux({
+  slice: 'autoswap',
+  initial: {
+    active: false,
+    connected: false,
+    account: null,
+    purses: null,
+    inputPurse: null,
+    outputPurse: null,
+    inputAmount: null,
+    outputAmount: null,
+    freeVariable: null,
+    invitationDepositId: null,
+  },
+  actions: {
+    activateConnection: state => ({ ...state, active: true }),
+    deactivateConnection: state => ({ ...state, active: false }),
+    serverConnected: state => ({ ...state, connected: true }),
```

Comment:
presumably there could be a separate reducer for e.g., `connected` vs. `active`?

Reply:
Not in the current architecture: there is just one context, and it has a single reducer.